A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,442 @@
+will be moving from its original grpclb protocol to the new xDS protocol.
+
+The xDS protocol allows configuring a variety of features, including
+many that gRPC does not currently support.  Over time, we will be slowly
+adding support for more of these features in gRPC.  We will publish a
+gRFC for each set of xDS features as we add support for them.
+
+This document describes the changes we are making to support global load
+balancing features, which is the set of xDS functionality that we are
+initially targetting.  This affects the gRPC client only; use of xDS in
+the gRPC server will come later as part of a separate set of xDS features.
+
+## Background
+
+The xDS API is actually a suite of APIs with names of the form "x
+Discovery Service", where there are many values for "x" (hence the name
+"xDS" for the overall protocol suite).  There is not currently a formal
+spec for the protocol, but the best reference is at
+https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol.
+Readers of this document should familiarize themselves with the xDS API
+before proceeding.
+
+The xDS APIs are essentially pub/sub mechanisms, where each API allows
+subscribing to a different type of configuration resource.  In the xDS
+API flow, the client uses the following main APIs, in this order:
+
+- __Listener Discovery Service (LDS)__: Returns
+  [`Listener`](https://github.com/envoyproxy/envoy/blob/9e83625b16851cdc7e4b0a4483b0ce07c33ba76b/api/envoy/api/v2/listener.proto#L27)
+  resources.  Used basically as a convenient root for the gRPC client's
+  configuration.  Points to the RouteConfiguration.
+- __Route Discovery Service (RDS)__: Returns
+  [`RouteConfiguration`](https://github.com/envoyproxy/envoy/blob/9e83625b16851cdc7e4b0a4483b0ce07c33ba76b/api/envoy/api/v2/route.proto#L24)
+  resources.  Provides data used to populate the gRPC [service
+  config](https://github.com/grpc/grpc/blob/master/doc/service_config.md).
+  Points to the Cluster.
+- __Cluster Discovery Service (CDS)__: Returns
+  [`Cluster`](https://github.com/envoyproxy/envoy/blob/9e83625b16851cdc7e4b0a4483b0ce07c33ba76b/api/envoy/api/v2/cluster.proto#L34)
+  resources.  Configures things like load balancing policy and load reporting.
+  Points to the ClusterLoadAssignment.
+- __Endpoint Discovery Service (EDS)__: Returns
+  [`ClusterLoadAssignment`](https://github.com/envoyproxy/envoy/blob/9e83625b16851cdc7e4b0a4483b0ce07c33ba76b/api/envoy/api/v2/endpoint.proto#L33)
+  resources.  Configures the set of endpoints (backend servers) to load
+  balance across and may tell the client to drop requests.
+
+gRPC will support the Aggregate Discovery Service (ADS) variant of xDS,
+where all of these resource types are obtained on a single gRPC stream.
+However, the different phases of the API flow are still typically
+referred to using the separate service names described above.
+
+In the future, we may add support for the incremental ADS variant of
```

Comment:
Are you planning on making any comments wrt v2 vs. v3 xDS now that this is a GA thing?